Annotato/Annotato/Persistence/DocumentsPersistenceManager.swift

```diff
@@ -60,25 +94,21 @@ class DocumentsPersistenceManager {
     }
 
     func updateDocument(document: Document) async -> Document? {
-        let remoteUpdatedDocument = await remoteDocumentsPersistence
-            .updateDocument(document: document)
+        _ = await remoteDocumentsPersistence.updateDocument(document: document)
 
-        if remoteUpdatedDocument == nil {
-            document.setUpdatedAt()
-        }
-
-        return localDocumentsPersistence.updateDocument(document: remoteUpdatedDocument ?? document)
+        document.setUpdatedAt()
+        return localDocumentsPersistence.updateDocument(document: document)
     }
 
     func deleteDocument(document: Document) async -> Document? {
-        let remoteDeletedDocument = await remoteDocumentsPersistence
-            .deleteDocument(document: document)
+        _ = await remoteDocumentsPersistence.deleteDocument(document: document)
 
-        if remoteDeletedDocument == nil {
-            document.setDeletedAt()
-        }
+        // NOTE: Documents are hard deleted from local storage
+        return localDocumentsPersistence.deleteDocument(document: document)
+    }
 
-        return localDocumentsPersistence.deleteDocument(document: remoteDeletedDocument ?? document)
+    func deleteDocumentLocally(document: Document) async -> Document? {
```

Comment:
```suggestion
    func deleteDocumentLocally(document: Document) -> Document? {
```

Can you also check for other local methods? Local methods should not be async.